crates/swc_ecma_parser/src/parser/pat.rs

```diff
@@ -707,36 +700,35 @@ impl<'a, I: Tokens> Parser<I> {
                     }
                 }
 
-                if count_of_trailing_comma == 0 {
-                    let expr = exprs.into_iter().next().unwrap();
-                    let last = match expr {
-                        // Rest
-                        Some(ExprOrSpread {
-                            spread: Some(dot3_token),
-                            expr,
-                        }) => {
-                            // TODO: is BindingPat correct?
-                            let expr_span = expr.span();
-                            self.reparse_expr_as_pat(pat_ty.element(), expr)
-                                .map(|pat| {
-                                    Pat::Rest(RestPat {
-                                        span: expr_span,
-                                        dot3_token,
-                                        arg: Box::new(pat),
-                                        type_ann: None,
-                                    })
+                let expr = exprs.into_iter().next().unwrap();
```

Comment:
Does this change depends on the `syntax_error!` above?

Reply:
No, parse_expr would eliminate tailing comma, so there's no trailing comma in reparse_expr_as_pat_inner, that's the block of #3676

Reply:
This change is wrong.